### PR TITLE
114 upgrade typescript and eslint

### PR DESCRIPTION
**Pull request by danieldunderfelt**
Closes https://github.com/HSLdevcom/bultti/issues/114

- Start using eslint and bring over same rules as in server
- Update Typescript
- Fix types throughout the app

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module.exports = {
+  ignorePatterns: ['!src/**/*', 'schema-types.ts'],
+  parser: '@typescript-eslint/parser', // Specifies the ESLint parser
+  parserOptions: {
+    ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
```
[the remark this answers is not in the document]
Comment (Divinare) [reply .eslintrc.js:5]: Tested having 2021, at least it works - didn't give any new errors

**Review by Divinare — CHANGES_REQUESTED** [review]
Well done! Nearly LGTM, left some questions

Good that we have now more typing and `any` is replaced with `unknown`

File: .eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (danieldunderfelt) [reply .eslintrc.js:5]: Sure, but it doesn't really matter as CRA is not really on the bleeding edge.

File: codegen.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 overwrite: true
+schema: 'schema.graphql'
```
[the remark this answers is not in the document]
Comment (danieldunderfelt) [reply codegen.yml:2]: Sure!

File: codegen.yml [hunk 3]
```diff
@@ -10,6 +10,9 @@ generates:
   ./src/schema-types.ts:
     plugins:
       - typescript
+      - typescript-operations
+    config:
+      defaultScalarType: 'string'
```
[the remark this answers is not in the document]
Comment (danieldunderfelt) [reply codegen.yml:15]: The default was `any`, now it is `string`. Much better, no?

We have only one custom scalar and it is a string. If we create more in the future which are not strings, `typescript-operation` allows for defining each scalar separately. I just don't see a reason to do that currently as the default works fine.

**Pull request merged**